File: src/go/9.palindrome-number.go
package main

func isPalindrome(x int) bool {
	if x < 0 {
		return false
	}
	orig := x
	rev := 0
	for x != 0 {
		rev = rev*10 + x%10
		x /= 10
	}
	return orig == rev
}

File: src/cpp/9.palindrome-number.cpp
/*
 * @lc app=leetcode id=9 lang=cpp
 *
 * [9] Palindrome Number
 */

// @lc code=start


// @lc code=start
#ifndef LC_LOCAL_TEST
class Solution {
public:
#endif
bool isPalindrome(int x) {
    if (x < 0) return false;
    int orig = x, rev = 0;
    while (x) {
        rev = rev * 10 + x % 10;
        x /= 10;
    }
    return orig == rev;
}
#ifndef LC_LOCAL_TEST
};
#endif
// @lc code=end
